service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -396,6 +397,39 @@ public String createAiNotebookInstance(
     return jobId;
   }
 
+  public ControlledAiNotebookInstanceResource updateAiNotebookInstance(
+      ControlledAiNotebookInstanceResource resource,
+      ApiGcpAiNotebookUpdateParameters updateParameters,
+      Optional<String> newName,
```

Comment:
generally `Optional` should only be used for return values. I think these should be `@Nullable String` instead

Reply:
why not? especially we don't have a static check for nullable